core/src/main/java/org/apache/servicecomb/core/SCBEngine.java

```diff
@@ -499,6 +500,23 @@ public MicroserviceReferenceConfig createMicroserviceReferenceConfig(String micr
     return createMicroserviceReferenceConfig(microserviceName, null);
   }
 
+  /**
+   * for edge, versionRule maybe controlled by url rule
```

Comment:
not only for edge, in the future, we should switch to this kind api for all cases.

Reply:
yes. but now only for edge. I find It's a huge work.